### PR TITLE
Allow for variable DQ/DQS ratio

### PR DESCRIPTION
**Pull request by RRozak**

### CODE REVIEW

File: litedram/phy/s7ddrphy.py [hunk 1]
```diff
@@ -50,9 +50,9 @@ def __init__(self, pads, with_odelay,
         bankbits    = len(pads.ba) if memtype != "DDR4" else len(pads.ba) + len(pads.bg)
         nranks      = 1 if not hasattr(pads, "cs_n") else len(pads.cs_n)
         databits    = len(pads.dq)
+        strobes     = len(pads.dqs_p)
         nphases     = nphases
         assert databits%8 == 0
-        x4_dimm_mode = (databits / len(pads.dqs_p)) == 4
```
Comment (acomodi) [inline litedram/phy/s7ddrphy.py:55]: we may have an assertion here that verifies that the `DQ/DQS` ratio is either `4` or `8`

Comment (acomodi) [reply litedram/phy/s7ddrphy.py:55]: Probably we can add it when the dq dqs ratio is defined below

Comment (RRozak) [reply litedram/phy/s7ddrphy.py:55]: Done

**Review by acomodi — APPROVED** [review]
LGTM. Let's open the PR upstream (or update the one already open)

**Comment by RRozak** [comment]
Thank you. I will try to fix the CI failures first.

**Comment by RRozak** [comment]
I see that these CI failures are also on our master, but not on enjoy-digital's master.

**Pull request closed without merging**